File: marvin_core/risk.py
from typing import Any


RISK_ORDER = {
    "low": 1,
    "medium": 2,
    "high": 3,
    "critical": 4,
}


def normalize_risk_level(value: Any) -> str:
    risk = str(value or "low").strip().lower()
    if risk not in RISK_ORDER:
        raise ValueError(f"Invalid risk level: {value}")
    return risk


def risk_meets_threshold(risk_level: str, threshold: str) -> bool:
    risk = normalize_risk_level(risk_level)
    minimum = normalize_risk_level(threshold)
    return RISK_ORDER[risk] >= RISK_ORDER[minimum]

File: dashboard-v2/lib/risk.ts
export type RiskLevel = "low" | "medium" | "high" | "critical" | "unknown";

export function normalizeRisk(value: string | null | undefined): RiskLevel {
  if (!value) return "unknown";
  const normalized = value.toLowerCase();
  if (normalized === "low" || normalized === "medium" || normalized === "high" || normalized === "critical") {
    return normalized;
  }
  return "unknown";
}

export function riskSeverity(risk: RiskLevel): number {
  const severity: Record<RiskLevel, number> = {
    low: 0,
    medium: 1,
    high: 2,
    critical: 3,
    unknown: -1
  };
  return severity[risk];
}
